modiscolite/meme_writer.py

```diff
@@ -0,0 +1,156 @@
+# Create a class that can be used to write MEME files.
+from typing import List, Optional
+import numpy as np
+
+from os import PathLike
+
+class MemeWriterMotif:
+	"""Class for handling Motif for MEME file writing."""
+
+	def __init__(
+		self,
+		name: str,
+		probability_matrix: np.ndarray,
+		# `alphabet_length` is known as 'alength' in MEME Suite.
+		alphabet_length: int,
+		# `source_sites` is known as 'nsites' in MEME Suite.
+		source_sites: int,
+		e_value: Optional[str] = None,
+		url: Optional[str] = None
+	) -> None:
+		self._name = name
+		self._probability_matrix = probability_matrix
+		self._alphabet_length = alphabet_length
+		self._source_sites = source_sites
+		self._e_value = e_value
+		self._url = url
+
+	@property
+	def name(self):
+		return self._name
+
+	@property
+	def probability_matrix(self):
+		return self._probability_matrix
+
+	@property
+	def alphabet_length(self):
+		return self._alphabet_length
+	
+	@property
+	def source_sites(self):
+		return self._source_sites
+
+	@property
+	def e_value(self):
+		return self._e_value
+
+	@property
+	def url(self):
+		return self._url
+
+	def __repr__(self) -> str:
+		return f"MemeWriterMotif(name={self._name})"
+
+
+class MemeWriter:
```

Comment:
It should be "MEME" for "Multiple Elucidation of Motif .. somethings", not meme like the images, which came later. So, `MEMEWriter`, though it might just make more sense to be a MEME object since it stores the motifs as well? Either way, it should be capitalized.

Reply:
We should probably also replace the MEME file reading/writing in `report.py` with this implementation after we confirm that it's working. Maybe let's get this merged in and confirmed and then separately handle that.

Reply:
I agree with using `MEME` rather than `Meme`- I just thought it was PEP convention to write acronums with CapWords, but after reading it again it appears not. Perhaps I was remembering a different standard

Reply:
Updated nonetheless